synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -87,8 +87,11 @@ def action_for_event_by_user(self, event, context):
         condition_cache = {}
 
         for uid, rules in self.rules_by_user.items():
-            display_name = room_members.get(uid, {}).get("display_name", None)
-            if not display_name:
+            display_name = None
+            profile_info = room_members.get(uid)
+            if profile_info:
+                display_name = profile_info.display_name
+            else:
```

Comment:
Is this else still correct? Could the member have a `profile_info.display_name == None`?